### DISCUSSION TITLE
user.Id = 0

### DISCUSSION THREAD

**Discussion by pedro-mileipp** [balta-io/forum#248] [Tira Dúvidas] [answered]
No módulo 4 do curso de Dapper, por que se coloca o user.Id = 0 no método abaixo?
```csharp
public void Create(User user, string connectionString){
    user.Id = 0;
    _connection.Insert<User>(user);
}
```

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @pedro-mileipp, se houver um "Id" diferente de zero, na hora de gerar o create ele não vai pegar um Id do banco automaticamente, causando erro.

Você pode testar para ver!

**Reply by oimarcella** [reply]
@andrebaltieri então para forçar a geração do id no Create, temos que definiri que será 0?
Isso é algo específicodo Identity?

**Reply by andrebaltieri** [reply]
Isso, mas toda propriedade `int` já vem com valor padrão como `0`.
Se você não quiser (ou esquecer de definir), tudo bem 😊
